src/query/ts/m3db/encoded_series_iterator.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3db
+
+import (
+	"math"
+	"sync"
+
+	"github.com/m3db/m3/src/dbnode/encoding"
+	"github.com/m3db/m3/src/dbnode/ts"
+	"github.com/m3db/m3/src/query/block"
+)
+
+type encodedSeriesIter struct {
+	mu                    sync.RWMutex
+	idx                   int
+	meta                  block.Metadata
+	seriesMeta            []block.SeriesMeta
+	consolidationSettings *consolidationSettings
+	seriesIters           []encoding.SeriesIterator
+}
+
+func makeEncodedSeriesIter(
+	meta block.Metadata,
+	seriesMeta []block.SeriesMeta,
+	seriesIters []encoding.SeriesIterator,
+) block.SeriesIter {
+	return &encodedSeriesIter{
+		idx:         -1,
+		meta:        meta,
+		seriesMeta:  seriesMeta,
+		seriesIters: seriesIters,
+	}
+}
+
+func (it *encodedSeriesIter) currentUnconsolidated() (block.Series, error) {
```

Comment:
instead of providing a view for consolidated vs unconsolidated at each iteration, it might be better to provide a consolidatedSeriesIter vs unconsolidatedSeriesIter and all their iterations are of the same kind

Reply:
Yeah this is a bit of a stopgap until your PR lands, then I'll change this encoded series iterator to satisfy both `block.Block` and `block.UnconsolidatedBlock`